data_structures/linked_list/circular_linked_list.py

```diff
@@ -0,0 +1,224 @@
+import typing
+
+
+class Node:
+    """
+    Class to represent a single node.
+
+    Each node has following attributes
+    * data
+    * next_ptr
+    """
+
+    def __init__(self, data: typing.Any):
+        self.data = data
+        self.next_ptr = None
+
+    def set_data(self, value: typing.Any):
+        """
+        Set the data field of the node to given value.
+        >>> node = Node(1)
+        >>> node.set_data(2)
+        >>> node.get_data()
+        2
+        """
+        self.data = value
+
+    def get_data(self) -> typing.Any:
+        """
+        Returns the data field of the current node.
+        >>> node = Node(1)
+        >>> node.get_data()
+        1
+        """
+        return self.data
+
+    def set_next(self, value: typing.Type['Node']):
+        """
+        Sets the next pointer of the current node.
+        >>> node, node1 = Node(1), Node(2)
+        >>> node.set_next(node1)
+        >>> node.get_next().get_data()
+        2
+        """
+        self.next_ptr = value
+
+    def get_next(self) -> typing.Type['Node']:
+        """
+        Returns the next pointer of the current node.
+        >>> node = Node(1)
+        >>> node.get_next() is None
+        True
+        """
+        return self.next_ptr
+
+    def has_next(self) -> bool:
+        """
+        Checks if the current node has a valid next node.
+        >>> node = Node(1)
+        >>> node.has_next()
+        False
```

Comment:
I just kept this method so that someone can use it later for adding more Higher level operations like, refactoring the `Node` class to provide direct access to member variables.